### PR TITLE
feat: Truncate and reduce label sizes + Responsive enhancements

### PR DESCRIPTION
**Pull request by dogmar**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203837229973050
  - https://app.asana.com/0/1203642829646384/1203896774152795/f

### CODE REVIEW

**Review by floreks — APPROVED** [review]
Just some linting issue that needs to be fixed, otherwise LGTM

File: assets/src/components/cluster/LabelsAnnotations.tsx [hunk 1]
```diff
@@ -34,45 +36,58 @@ export function LabelsAnnotationsRow({
...
-} & CardProps) {
+  } & CardProps) {
+  const theme = useTheme()
+
   return (
-    <Card
-      padding="xlarge"
-      {...props}
-    >
+    <MetadataCard {...props}>
```
Comment (maciaszczykm) [inline assets/src/components/cluster/LabelsAnnotations.tsx:64]: The card will be empty if there are no labels and annotations?

Comment (dogmar) [reply assets/src/components/cluster/LabelsAnnotations.tsx:64]: Yeah, I guess I should hide the card completely in that case.

Comment (dogmar) [reply assets/src/components/cluster/LabelsAnnotations.tsx:72]: Doesn't render card if there's no data at all.

**Pull request merged**